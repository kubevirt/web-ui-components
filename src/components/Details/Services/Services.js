import React from 'react';
import PropTypes from 'prop-types';

import { getName, getNamespace } from '../../../selectors';
import { getServicesForVm } from '../../../selectors/service';
import { InlineLoading } from '../../Loading';
import { DASHES } from '../../../constants';
import { ServiceModel } from '../../../models';

export const Services = ({ services, vm, ResourceLinkComponent, LoadingComponent }) => {
  if (!services) {
    return <LoadingComponent />;
  }
  const vmServices = getServicesForVm(services, vm);
  return vmServices.length === 0
    ? DASHES
    : vmServices.map(service => (
        <ResourceLinkComponent kind={ServiceModel.kind} name={getName(service)} namespace={getNamespace(service)} />
      ));
};

Services.defaultProps = {
  services: null,
  LoadingComponent: InlineLoading,
};

Services.propTypes = {
  services: PropTypes.array,
  vm: PropTypes.object.isRequired,
  ResourceLinkComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
};
