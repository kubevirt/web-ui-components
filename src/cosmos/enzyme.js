import createContext from 'react-cosmos-test/enzyme';

import paths from '../../config/paths';

export default function createTestContext(args) {
  args.proxies = args.proxies || []; // workaround of react-cosmos-test bug
  return createContext({
    ...args,
    cosmosConfigPath: `${paths.config}/cosmos.config.js`,
  });
}
