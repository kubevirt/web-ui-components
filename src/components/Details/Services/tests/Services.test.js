import React from 'react';
import { shallow } from 'enzyme';

import { Services } from '../Services';

import { default as servicesFixtures } from '../fixtures/Services.fixture';

// eslint-disable-next-line react/prop-types
const testServices = ({ props }) => <Services {...props} />;

describe('<Services />', () => {
  it('renders correctly', () => {
    servicesFixtures.forEach(fixture => {
      const component = shallow(testServices(fixture));
      expect(component).toMatchSnapshot();
    });
  });
});
