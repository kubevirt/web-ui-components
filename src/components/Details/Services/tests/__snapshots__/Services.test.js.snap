// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Services /> renders correctly 1`] = `
Array [
  <ResourceLinkComponent
    kind="Service"
    name="fooService"
  />,
  <ResourceLinkComponent
    kind="Service"
    name="fooService1"
  />,
]
`;

exports[`<Services /> renders correctly 2`] = `"---"`;

exports[`<Services /> renders correctly 3`] = `
<InlineLoading
  size="md"
/>
`;
